compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1524,13 +1531,15 @@ fn link_local_crate_native_libs_and_dependent_crate_libs<'a, B: ArchiveBuilder<'
     // If -Zlink-native-libraries=false is set, then the assumption is that an
     // external build system already has the native dependencies defined, and it
     // will provide them to the linker itself.
+    cmd.group_start();
     if sess.opts.debugging_opts.link_native_libraries {
         add_local_native_libraries(cmd, sess, codegen_results);
     }
     add_upstream_rust_crates::<B>(cmd, sess, codegen_results, crate_type, tmpdir);
     if sess.opts.debugging_opts.link_native_libraries {
         add_upstream_native_libraries(cmd, sess, codegen_results, crate_type);
     }
+    cmd.group_end();
```

Comment:
This allows native libs linking to rust libs. Do we want this or could each kind of library (local native, upstream rust, upstream native) get it's own group?

Reply:
@bjorn3 In some cases it should be possible to resolve symbols across those groups. For instance, compiler-builtins in Rust provides symbols that any of the other groups may need.